src/components/formComponentTemplate.tsx

```diff
@@ -421,6 +509,69 @@ export const formComponentTemplate = (
         ),
         validate: validate.publicKey,
       };
+    case "source_account":
+      return {
+        render: (templ: TemplateRenderProps) => (
+          <PubKeyPicker
+            key={id}
+            id={id}
+            label="Source Account"
+            labelSuffix={!templ.isRequired ? "optional" : undefined}
+            value={templ.value || ""}
+            error={templ.error}
+            onChange={templ.onChange}
+          />
+        ),
+        validate: validate.publicKey,
+      };
+    case "source_amount":
+      return {
+        render: (templ: TemplateRenderProps) => (
+          <TextPicker
+            key={id}
+            id={id}
+            label="Source Amount"
+            labelSuffix={!templ.isRequired ? "optional" : undefined}
+            value={templ.value || ""}
+            error={templ.error}
+            onChange={templ.onChange}
+          />
+        ),
+        validate: validate.amount,
+      };
+    case "source_asset":
+      return {
+        render: (templ: TemplateRenderAssetProps) => (
+          <AssetPicker
+            key={id}
+            assetInput={custom?.assetInput}
+            id="destination_asset"
```

Comment:
I think this is supposed to be either `{id}` or `source_asset`

Reply:
Yeah, I'll update to use `id` in components where possible to avoid typos like this.